packages/machine-extractor/src/types.ts

```diff
@@ -38,4 +78,64 @@ export interface FileExtractResult<
   file: t.File;
 }
 
-export type DeclarationType = 'named' | 'inline' | 'identifier' | 'unknown';
+export type DeclarationType =
+  | 'named'
+  | 'inline'
+  | 'identifier'
+  | 'object'
+  | 'unknown';
+
+type BaseBuiltinAction<P extends object> = {
+  kind: 'builtin';
+  action: { type: string } & P;
+};
+export type ExtractorAssignAction = BaseBuiltinAction<{
+  assignment:
+    | Record<string, JsonItem | JsonExpressionString>
+    | JsonExpressionString;
+}>;
+export type ExtractorRaiseAction = BaseBuiltinAction<{
+  event: Record<string, JsonItem> | JsonExpressionString;
+}>;
+export type ExtractedLogAction = BaseBuiltinAction<{
+  expr: string | JsonExpressionString;
```

Comment:
this is equivalent of `string` - so the q is: what value does it bring us here? it has some documentation value of the intent but one that's only visible in this file, if we hover over those properties in different places we won't see it

Reply:
Just educational. I'd say when we later look at the source of these type definitions, it's valuable to still see if a string can be an expression string too or not.